### PR TITLE
Not all parents have a fire method

### LINKED ISSUE #8: Exception in detached
Sorry I didn't catch this in the PR. `this._parentForm` is set to the parent node, which may or may not be a Polymer element and might not define the `fire` function.

### PR DESCRIPTION
**Pull request by notwaldorf**
Fixes https://github.com/PolymerElements/iron-form-element-behavior/issues/8

(I `bower-link`ed this locally and it fixes the `paper-input` tests)

### CODE REVIEW

**Comment by hadfieldn** [comment]
@notwaldorf Just ran into this issue with the parent node not having a fire method -- thanks for working on a fix. 

The solution proposed in this PR still assumes that the parent node is the form, but it seems likely that the element could be wrapped in something else before it's wrapped in the form (or it might not be wrapped in a form at all). I'm wondering a reference to the parent form is really needed. Could  `attached()` and `detached()` call `Polymer.Base.fire()` directly, and alleviate the need for keeping a reference to the parent form?

``` js
    attached: function() {
      Polymer.Base.fire('iron-form-element-register', null, {node:this});
    },

    detached: function() {
      Polymer.Base.fire('iron-form-element-unregister', {target: this}, {node:this});
    }
```

**Comment by notwaldorf** [comment]
@hadfield I originally tried that and it didn't work. My understanding is that `detached` is called after the child element has been removed, so it can fire an event all it wants, but there's nobody to listen to it.

**Comment by notwaldorf** [comment]
@morethanreal ping! 🐼

**Comment by morethanreal** [comment]
It might be possible to have the parent form set a property on the form element to itself in its handler for `iron-form-element-register`, so the form element will have a reference to the actual form.

**Comment by notwaldorf** [comment]
@morethanreal oh yeah, that's way better. herp derp. PTAL?

File: iron-form-element-behavior.html [hunk 1]
```diff
@@ -52,22 +52,20 @@
       },
 
       /**
-       * Need to keep a reference to the form this element is registered
-       * to, so that it can unregister if detached.
+       * The form that the element is registered to.
        */
-      _parentForm: {
+      parentForm: {
```
Comment (morethanreal) [inline iron-form-element-behavior.html]: Might be ok to keep it private since it's only used to coordinate between this behavior and the form behavior?

**Comment by morethanreal** [comment]
lgtm

**Comment by notwaldorf** [comment]
Alright, renamed it back to `_parentForm` and also added a comment about when the property is actually being set. Thanks! 😻

**Pull request merged**